src/components/pages/Home/Home.tsx

```diff
@@ -1,19 +1,62 @@
 import SearchBar from '../../inputs/Search/SearchBar/SearchBar';
 import './styles.css';
+import { useStateValue } from '../../../state/state';
+import { useState, useEffect } from 'react';
+import AvailabilityHeader from '../../layout/AvailabilityHeader/AvailabilityHeader';
 
 function Home() {
+  const [{ arnsSourceContract }] = useStateValue();
+  const [records, setRecords] = useState({});
+
+  const [availability, setAvailability] = useState('search');
+  const [searchName, setSearchName] = useState('');
+
+  const [placeholderText, setPlaceholderText] = useState('Find a name');
+
+  useEffect(() => {
+    let newRecords = arnsSourceContract.records;
+    setRecords(newRecords);
+  }, [arnsSourceContract]);
+
+  function availiblityCheck(props: string) {
+    // if name is not in the legal character range or chars, dnt run
+    const namePattern = new RegExp('^[a-zA-Z0-9_-]{1,32}$');
+    if (!namePattern.test(props) && props !== '') {
+      console.log('not a valid name', namePattern.test(props));
+      return;
+    } else {
+      //if empty reset to search state
+      if (props == '') {
+        setAvailability('search');
+        setPlaceholderText('Find a name');
+      }
```

Comment:
we should avoid even getting to this point, and validate that the supplied text is not an empty string before calling this function